### PR TITLE
kafka 0.8.2.0

### PR DESCRIPTION
**Pull request by mikeclarke**
- Upgrade to latest stable kafka server
- Regenerate the gradle wrapper before building the jar
- Remove the number of partitions replace; `num.partitions=1` is now the default
- Update installed folders to match upstream refactoring
- Remove warning about Java 7, latest stable supports both Java 7 and 8

### CODE REVIEW

File: Library/Formula/kafka.rb [hunk 1]
```diff
@@ -27,10 +26,13 @@ def install
...
 
-    libexec.install %w(contrib core examples lib perf system_test)
+    # Workaround for conflicting slf4j-log4j12 jars (1.7.6 is preferred)
+    rm_f "core/build/dependant-libs-2.10.4/slf4j-log4j12-1.6.1.jar"
+
+    libexec.install %w(clients contrib core examples system_test)
 
     prefix.install "bin"
-    bin.env_script_all_files(libexec/"bin", :JAVA_HOME => "`/usr/libexec/java_home`")
+    bin.env_script_all_files(libexec/"bin", :JAVA_HOME => "${JAVA_HOME:=`/usr/libexec/java_home`}")
```
Comment (mikemcquaid) [inline Library/Formula/kafka.rb]: Hmm, not sure what I think about this given the previous version is what we do for pretty much all other software.

File: Library/Formula/kafka.rb [hunk 2]
```diff
@@ -3,18 +3,17 @@
...
+  depends_on "gradle"
   depends_on "zookeeper"
   depends_on :java => "1.7"
 
   def install
-    system "./gradlew", "jar"
-
-    # Use 1 partition by default so individual consumers receive all topic messages
-    inreplace "config/server.properties", "num.partitions=2", "num.partitions=1"
+    system "gradle"
```
Comment (mikemcquaid) [inline Library/Formula/kafka.rb]: Any idea why this is needed now?

File: Library/Formula/kafka.rb (hunk 1 above)
Comment (mikeclarke) [inline Library/Formula/kafka.rb]: Sounds good - I'll revert.

File: Library/Formula/kafka.rb (hunk 2 above)
Comment (mikeclarke) [inline Library/Formula/kafka.rb]: KAFKA-1490 removed the gradle jar from kafka source and introduced the dependency on gradle. The 0.8.2 readme was updated to include the additional steps - https://github.com/apache/kafka/blob/0.8.2/README.md.

Comment (mikemcquaid) [inline Library/Formula/kafka.rb]: :+1:

**Pull request closed without merging**